docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,420 @@
+
+> [!NOTE]
+> It can be cumbersome to manually define test classes, especially if you need to customize behavior differently for each test. Consider using a library like Moq to streamline your testing.
+
+```csharp
+public class MockSecretClient : SecretClient 
+{
+    public override Response<KeyVaultSecret> GetSecret(
+        string name,
+        string version = null, 
+        CancellationToken cancellationToken = default)
+        => throw new NotImplementedException();
+        
+    public override Task<Response<KeyVaultSecret>> GetSecretAsync(
+        string name, 
+        string version = null, 
+        CancellationToken cancellationToken = default)
+        => throw new NotImplementedException();
+}
+
+SecretClient secretClient = new MockSecretClient();
+```
+
+---
+
+### Service client input and output models
+
+Model types hold the data being sent and received from Azure services. There are two main kinds of models:
+
+* **Input models** are intended to be created and passed as parameters to service methods by developers. They have one or more public constructors and writeable properties.
+* **Output models** are only returned by the service and have neither public constructors nor writeable properties.
+
+To create a test instance of an input model use one of the available public constructors and set the additional properties you need.
+
+```csharp
+SecretProperties secretProperties = new("secret")
+{
+    NotBefore = DateTimeOffset.Now
+};
+```
+
+To create instances of output models, a model factory is used. Most Azure SDK client libraries provide a static model factory class that ends in `ModelFactory` and contains a set of static methods to initialize the library's output model types.
+
+```csharp
+KeyVaultSecret keyVaultSecret = SecretModelFactory.KeyVaultSecret(
+    new SecretProperties("secret"), "secretValue");
+```
+
+> [!NOTE]
+> Some input models have read-only properties that are only populated when the model is returned by the service. In this case, a model factory method will be available that allows setting these properties. See <xref:Azure.Security.KeyVault.Secrets.SecretModelFactory>.
+
+```csharp
+// CreatedOn is a read-only property and can only be 
+// set via a model factory SecretProperties.
+secretPropertiesWithCreatedOn = SecretModelFactory.SecretProperties(
+    name: "secret", createdOn: DateTimeOffset.Now);
+```
+
+## Explore response types
+
+The <xref:Azure.Response> class is an abstract class that represents an HTTP response and is returned by most service client methods. You can create test `Response` instances using either the Moq library or standard C# inheritance.
+
+## [Moq](#tab/moq)
+
+The Moq library provides concise functionality for setting up mock responses:
+
+```csharp
+Mock responseMock = new Mock();
```

Comment:
```suggestion
Mock<Response> responseMock = new Mock<Response>();
```
The `new` won't compile without the generic included.  If you use `Mock` as the type (sans generic) then you'll need to do an explicit cast on L124

```csharp
Response response = (Response)responseMock.Object;
```